File: Problem Set 2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string para = get_string("The text: ");

    //main algorithm
    int numWords = 1;
    int numLetters = 0;
    int numSentences = 0;

    for(int i = 0; i < strlen(para); i++){
        //count of letters
        if((int) para[i] >= 65 && (int) para[i] <= 122){
            numLetters++;
            }

        if((int) para[i] >= 91 && (int) para[i] <= 96){
            numLetters = numLetters - 1;
            }

        //count of sentences
        if((int) para[i] == 33 || (int) para[i] == 46 || (int) para[i] == 63){
            numSentences++;
            }

        //count of words
        if( ((int) para[i] == 34 || (int) para[i] == 40 || (int) para[i] == 32) && (int) para[i+1] >= 65 && (int) para[i+1] <= 122){
            numWords++;
            }
        }

    //calculating the index
    float L = (numLetters * 100) / numWords;
    float S = (numSentences * 100) / numWords;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    //printing out result
    if(index <= 1){
        printf("Before Grade 1\n");
    }

    else if(index > 16){
        printf("Grade 16+\n");
    }

    else{
        printf("Grade %.0f\n", index);
        //printf("%i%i%i", numLetters, numWords, numSentences);
    }
}

File: Problem Set 6/readability.py
from cs50 import get_string

text = get_string("Text: ")
numLetters = 0
numWords = 1
numSentences = 0
length = len(text)

#Counting the num of lettters, words and sentences
for i in range(length):
    #Count of number of letters in the text
    if ord(text[i]) >= 65 and ord(text[i]) <= 122:
        numLetters += 1
        
    if ord(text[i]) >= 91 and ord(text[i]) <= 96:
        numLetters -= 1
    
    #Count for number of words in the text
    if ord(text[i]) == 34 or ord(text[i]) == 40 or ord(text[i]) == 32 and ord(text[i+1]) >= 65 and ord(text[i+1]) <= 122:
        numWords += 1
    
    #Count of number of sentences in the text
    if ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63:
        numSentences += 1

#Calculating the index
L = float((numLetters * 100) / numWords)
S = float((numSentences * 100) / numWords)
index = round((0.0588 * L) - (0.296 * S) - 15.8)

#printing out result
if index <= 1:
    print("Before Grade 1")

elif index > 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")

print(f"{numLetters} , {numWords}, {numSentences}")